hardware_interface/source/adcs.c

```diff
@@ -184,7 +184,7 @@ ADCS_returnState HAL_ADCS_get_TC_ack(ADCS_TC_ack *TC_ack) {
     #endif
 }
 
-ADCS_returnState HAL_ADCS_get_file_download_buffer(uint16_t *packet_count, uint8_t **file[20]) {
+ADCS_returnState HAL_ADCS_get_file_download_buffer(uint16_t *packet_count, uint8_t file[20]) {
```

Comment:
Is it valid in C to pass an array like this?

Reply:
Usually you don't specify the array size on a parameter list, i.e. everyone else would use "uint8_t *file". I believe this actually gives a bit of length checking, in that it would be illegal to pass "uint_8 too_short[10]". But it's also illegal to pass the "(uint8_t *) malloc(20)", and it's confusing because an array is an address (as is a function) so this looks like call by value but I believe the "value" is the address. My actual preference would be to rewrite it to uint8_t* and include some unit tests to prove that it works - for now I just wanted to get rid of the compiler warnings.